tenseal/cpp/tensors/ckksvector.cpp

```diff
@@ -570,6 +570,60 @@ CKKSVector& CKKSVector::polyval_inplace(const vector<double>& coefficients) {
     return *this;
 }
 
+CKKSVector CKKSVector::conv2d_im2col(const vector<double>& kernel,
+                                     size_t windows_nb) {
+    CKKSVector new_vec = *this;
+    new_vec.conv2d_im2col_inplace(kernel, windows_nb);
+    return new_vec;
+}
+
+CKKSVector& CKKSVector::conv2d_im2col_inplace(const vector<double>& kernel,
+                                              size_t windows_nb) {
+    vector<double> plain_vec;
+    size_t chunck_size = windows_nb;
```

Comment:
I think we can simplify the API by passing a real kernel (matrix) and pad it inside the function, based on the chunk_size.

Reply:
Variable names are also confusing, chunk_size and windows_nb seems to be the same thing, why are we using both of them here, what does the API should have instead?

Reply:
Yeah, you are right, we simplify the API in that way, I'm planning to do that in another PR, as well as padding and applying im2col on the input matrix.